### ISSUE TITLE
Resource temporarily unavailable (os error 11) when sending stream to SPARQL store

### ISSUE THREAD

**Issue by tpluscode** [oxigraph/oxigraph#1118]
**Describe the bug**

Trying to send a SPARQL Store streaming request from node (fetch API) doesn't work

**To Reproduce**
1. Running oxigraph 0.4.2 from docker image
2. Run this code (node 20.17)
```js
import { Readable } from 'node:stream'

const triples = '_:a <http://example.org/p> "o" .'

const body = new Readable({
  read() {
    this.push(triples)
    this.push(null)
  },
})

const storeUrl = new URL('http://localhost:7878/store')
storeUrl.searchParams.set('graph', 'http://example.org/g')

;(async () => {
  const res = await fetch(storeUrl, {
    method: 'POST',
    headers: { 'content-type': 'application/n-triples' },
    body,
    duplex: 'half',
  })

  console.log(await res.text())
})()
```

The request takes a long time and the returns a 500 response saying `Resource temporarily unavailable (os error 11)`

Works fine when I send a preserialized payload as string instead

**Comment by Tpt** [comment]
Thank you for the bug report! This was a fun bug to track down: the NTriples parser is sometime calling an extra time the `read()` method on the HTTP response body even after it returned EOF. And the HTTP library chunked transfer encoding parser had a bug that made it try to read more from the socket even after reading the end of body data.

It wrote a fix for it that I plan to merge soon https://github.com/oxigraph/oxhttp/pull/38

**Comment by tpluscode** [comment]
> This was a fun bug to track down

I'm glad I provided entertainment :D

And as always, thank you for acting so quick!

**Comment by Tpt** [comment]
> I'm glad I provided entertainment :D

I signed up for this kind of issues when writing and using my own HTTP library

**Comment by Tpt** [comment]
The fix is shipped in v0.4.5. Feel free to reopen if it's not working properly.
